src/test/finalization/state_processor_tests.cpp

```diff
@@ -22,21 +22,21 @@ class Fixture {
             &m_finalization_params, &m_block_indexes, &m_chain, &m_state_db, &m_block_db)),
         m_proc(finalization::StateProcessor::New(
             &m_finalization_params, m_repo.get(), &m_chain)) {
-    m_chain.stub_AtHeight = [this](blockchain::Height h) -> CBlockIndex * {
+    m_chain.mock_AtHeight.SetStub([this](blockchain::Height h) -> CBlockIndex * {
       auto const it = this->m_block_heights.find(h);
       if (it == this->m_block_heights.end()) {
         return nullptr;
       }
       return it->second;
-    };
+    });
   }
 
   CBlockIndex &CreateBlockIndex() {
```

Comment:
My original intention was to streamline the mocks, I touched these tests because I had to. While I'd happily change to using `BlockIndexFake` I'd tend to follow up on this in a follow-up pull request.

WDYT?

Reply:
Yeah sure, I think I also can adapt these test further when I touch them next time.